### ISSUE TITLE
DOC501 doesn't trigger if a custom exception imported from another module is raised

### ISSUE THREAD

**Issue by taranlu-houzz** [astral-sh/ruff#13383]
The `DOC501` lint rule catches a custom exception missing in the docstring if the exception class is in the same module, but fails to catch it if the exception is imported from a different module.

The following test screenshot illustrates this:
![image](https://github.com/user-attachments/assets/c3ee1373-87cf-4be8-b0ed-973c29a8e39f)

```python
"""A module to illustrate ruff issue with custom exceptions."""

from pathlib import Path

from . import exceptions as E  # noqa: N812


class MyCustomError(Exception):
    """A custom exception class."""

    def __init__(self, message: str):
        self.message = message
        super().__init__(message)


def test_function() -> None:
    """Raise exception defined in same file."""
    raise MyCustomError("This is a custom error.")


def test_function2() -> None:
    """Raise exception defined in another module."""
    raise E.DuplicateTileDataInPerforceError({"pid": [Path("/test"), Path("/test2")]})
```

Version: `ruff 0.6.5`
Settings:
```toml
####################################################################################################
# Ruff
####################################################################################################

[tool.ruff]
line-length = 100
target-version = "py311"

[tool.ruff.lint]
ignore = [
  "B008", # do not perform function calls in argument defaults
  "C901", # too complex
  "E501", # line too long, handled by black
]
preview = true
select = [
  "B", # flake8-bugbear
  "C4", # flake8-comprehensions
  "D", # pydocstyle
  "DOC", # pydoclint
  "E", # pycodestyle errors
  "F", # pyflakes
  "FAST", # fastapi
  "FLY", # flynt
  "FURB", # refurb
  "I", # isort
  "N", # pep8-naming
  "PERF", # perflint
  "RUF", # ruff
  "UP", # pyupgrade
  "W", # pycodestyle warnings
]

[tool.ruff.lint.pydocstyle]
convention = "numpy"

[tool.ruff.lint.isort]
force-single-line = true
lines-after-imports = 2
```

**Comment by charliermarsh** [comment]
I think this does work correctly. Given `foo/bar.py`:

```python
from . import Exceptions as E

def func():
    """abc def"""
    raise E.Bar()
```

I get:

```
foo/bar.py:5:11: DOC501 Raised exception `Bar` missing from docstring
  |
3 | def func():
4 |     """abc def"""
5 |     raise E.Bar()
  |           ^^^^^^^ DOC501
  |
  = help: Add `Bar` to the docstring
```

What you might be seeing is that if you put `from . import Exceptions as E` in a file that isn't in a Python module (like, a directory without an `__init__.py`), then we indeed can't resolve the import, so the violation won't be flagged.

**Comment by taranlu-houzz** [comment]
@charliermarsh Hmm, possibly? My example is a cutdown version of what I actually ran into. The actual project where I noticed this initially has a structure like this:
```
.
├── src
│  └── A
│     └── B
│        └── C
│           └── D
│              ├── _private
│              │  ├── exceptions.py
│              │  ├── lib.py
│              │  └── logging.py
│              └── __init__.py
├── pyproject.toml
└── uv.lock
```

**Comment by charliermarsh** [comment]
Can you come up with an MRE that includes a valid package structure, like above? (As opposed to a standalone script in the initial comment.)

**Comment by taranlu-houzz** [comment]
Yes, I will try to put that together soon.

**Comment by MichaReiser** [comment]
I'll close this because it still  needs an mre. @taranlu-houzz feel free to comment with an mre and I'll reopen

**Comment by taranlu-houzz** [comment]
Apologies for not getting the mre together. I can't make this a priority right now, but if I do run into the issue again, I will try to do that. Thanks!
